lib/openssl/buffering.rb

```diff
@@ -398,7 +398,7 @@ def <<(s)
   # See IO#puts for full details.
 
   def puts(*args)
-    s = ""
+    s = String.new
```

Comment:
You'll probably want to use `+""` as `String.new` has different encoding

Reply:
Thanks I didn’t know it. Should we use encoding of output buffer? Maybe binary?

Reply:
Not sure if there's something else that might be better - but at least this way you keep the original behaviour:
```
irb(main):001:0> (+"").encoding
=> #<Encoding:UTF-8>
irb(main):002:0> String.new.encoding
=> #<Encoding:ASCII-8BIT>
```

Reply:
I would be concerned about using UTF-8 for a network buffer.